mustache_lint/mustache_lint.sh

```diff
@@ -24,10 +24,12 @@ mydir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 validator=${validator:-"https://html5.validator.nu"}
 if [ -f "$mydir/../node_modules/vnu-jar/build/dist/vnu.jar" ]
 then
-    if java -version 2>&1 >/dev/null | grep -q "java version"
+    if java -version 2>&1 >/dev/null | grep -Eq "(java|openjdk) version"
     then
-        javaversion=$(java -version 2>&1 | awk -F '"' '/version/ {print $2}')
-        if [[ "$javaversion" > "1.8" ]]
+        # Extract major and minor digits of version and compare it as float
+        # numbers. Not error proof, but sufficient to ensure it is >= 1.8
+        javaversion=$(java -version 2>&1 | awk -F '"' '/version/ {print $2}' | sed 's/^\([0-9]*\.[0-9]*\).*$/\1/')
+        if [ $( echo "${javaversion} >= 1.8" | bc ) == 1 ]
```

Comment:
Won't this fail the day enter into the 2 digit comparisons?

```echo "1.10 >= 1.8" | bc``` = 0

Reply:
Yep, that is why I state it is sufficient, but not error proof :) There is no java 1.10 in such  notation (to the best of my knowledge). Java 10 in Oracle versioning was [18.3](https://www.jcp.org/en/jsr/detail?id=383) (they switched to year/month based versioning since version 10), I can't check how that looks in installation, but if openjdk package is used, it reads 11.x (there are no [Debian](https://packages.debian.org/search?keywords=openjdk)/Ubunitu packages for anything between 1.8  and 11). So, I think for purpose of ensuring that version is higher than 1.8, the suggested comparison is sufficient (at least until we find an edge case where this does not work 😄 )

Reply:
Alternatively, it is possible to compare that version is not lower than 1.8, though not sure if that will make it more error proof. What do you think?

Reply:
Nah, all ok. If some day there is a situation when it fails... we can amend it, no big problem.